queries/odin/rainbow-delimiters.scm

```diff
@@ -0,0 +1,83 @@
+(parameters
+  "(" @delimiter
+  ")" @delimiter @sentinel) @container
+
+(tuple_type
+  "(" @delimiter
+  ")" @delimiter @sentinel) @container
+
+(parenthesized_expression
+  "(" @delimiter
+  ")" @delimiter @sentinel) @container
+
+(polymorphic_parameters
+  "(" @delimiter
+  ")" @delimiter @sentinel) @container
+
+(polymorphic_type
+  "(" @delimiter
+  ")" @delimiter @sentinel) @container
+
+(attribute
+  "(" @delimiter
+  ")" @delimiter @sentinel) @container
+
+(call_expression
+  "(" @delimiter
+  ")" @delimiter @sentinel) @container
+
+(index_expression
+  "[" @delimiter
+  "]" @delimiter @sentinel) @container
+
+(slice_expression
+  "[" @delimiter
+  "]" @delimiter @sentinel) @container
+
+(block
+  "{" @delimiter
+  "}" @delimiter @sentinel) @container
+
+(switch_statement
+  "{" @delimiter
+  "}" @delimiter @sentinel) @container
+
+(array_type
+  "[" @delimiter
+  "]" @delimiter @sentinel) @container
+
+(struct
+  ["{" "(" "["] @delimiter
+  ["}" ")" "]"] @delimiter @sentinel) @container
```

Comment:
This does not highlight correctly. How about this?

```suggestion
(struct
  ("(" @delimiter
   ")" @delimiter)?
  ("[" @delimiter
   "]" @delimiter)?
  "{" @delimiter
  "}" @delimiter @sentinel) @container

```

It assumes that the curly braces are mandatory and the parentheses and square brackets are optional.